torchrl/data/tensordict/tensordict.py

```diff
@@ -4732,3 +4732,37 @@ def _expand_to_match_shape(parent_batch_size, tensor, self_batch_dims, self_devi
 #
 #
 # _register_pytree_node(TensorDict, _flatten_tensordict, _unflatten_tensordict)
+
+
+def make_tensordict(
+    source: Union[TensorDictBase, dict],
+    batch_size: Optional[Union[Sequence[int], torch.Size, int]] = None,
+    device: Optional[DEVICE_TYPING] = None,
+) -> TensorDict:
+    """Returns a TensorDict created from the input source.
+    If batch_size is not specified, returns the maximum batch size possible
+
+    Args:
+        source (TensorDict or dictionary): a data source.
+        batch_size (iterable of int, optional): a batch size for the tensordict.
+        device (torch.device or compatible type, optional): a device for the TensorDict.
+    """
+    if batch_size is None:
+        batch_size = _find_max_batch_size(source)
+    return TensorDict(source, batch_size=batch_size, device=device)
+
+
+def _find_max_batch_size(source: Union[TensorDictBase, dict]) -> list[int]:
+    tensor_data = [tensor for _, tensor in source.items()]
```

Comment:
what happens if we have an empty list here?

Reply:
you can also call `values()` instead of `items()`

Reply:
good catch